server/src/main/java/org/elasticsearch/index/engine/ElasticsearchConcurrentMergeScheduler.java

```diff
@@ -132,6 +133,8 @@ protected void doMerge(MergeSource mergeSource, MergePolicy.OneMerge merge) thro
         try {
             beforeMerge(onGoingMerge);
             super.doMerge(mergeSource, merge);
+        } catch (AlreadyClosedException ignore) {
+            // Intentially ignored in order not to fail `IndexWriter` when we already shutting down the service.
```

Comment:
I wonder if we should have at least a `logger.debug` here.It proved to be helpful while debugging unexpected segment count